device_doctor/lib/src/device.dart

```diff
@@ -4,22 +4,36 @@
 
 import 'dart:async';
 
+import 'android_device.dart';
 import 'health.dart';
 import 'ios_device.dart';
+import 'utils.dart';
 
 /// Discovers available devices and chooses one to work with.
 abstract class DeviceDiscovery {
   factory DeviceDiscovery(String deviceOs) {
     switch (deviceOs) {
       case 'ios':
+        for (String property in supportedIosProperties) {
+          if (!properties.keys.contains(property)) {
+            throw FormatException('\n-----\n$property is not defined for $deviceOs device.\n-----');
+          }
+        }
         return IosDeviceDiscovery();
+      case 'android':
+        for (String property in supportedAndroidProperties) {
+          if (!properties.keys.contains(property)) {
+            throw FormatException('\n-----\n$property is not defined for $deviceOs device.\n-----');
+          }
+        }
+        return AndroidDeviceDiscovery();
       default:
         throw StateError('Unsupported device operating system: $deviceOs');
     }
   }
 
   /// Lists all available devices' IDs.
-  Future<List<Device>> discoverDevices({Duration retriesDelayMs = const Duration(seconds: 10)});
+  Future<List<Device>> discoverDevices({Duration retriesDelay = const Duration(seconds: 10)});
```

Comment:
retriesDelaySeconds?

Reply:
Done.

Reply:
Durations can be minutes, hours, and days too. We shouldn't be specifying the time unit in the name. What about `retryDuration`?

Reply:
Here, and also in this tool, we are using seconds, which makes it clear about the delay. Any doc suggests not specifying the unit?

Reply:
I can't find anything specific in style guides, but searching the Flutter repo shows this is followed there: https://cs.opensource.google/search?q=Duration&sq=&ss=flutter%2Fflutter

Reply:
Looking through list of flutter repo, I agree that it makes sense to avoid the unit in the name. Updated it with suggested `retryDuration`.